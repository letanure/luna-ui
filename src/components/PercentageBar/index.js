import React from 'react'

const PercentageBar = ({ values, width, fontSize, padding = '1px 2px' }) => {
  const tenPercentOfFullWidth = width * 0.1
  const minWidth = tenPercentOfFullWidth > 20 ? tenPercentOfFullWidth : 20
  const total = values.reduce((acc, value) => acc += value.amount, 0)
  return (
    <div style={{
      width,
      display: 'flex'
    }}>
      {
        values.map(value => {
          const relativeSize = (value.amount * 100) / total
          const partialWidth =  width * (relativeSize / 100)
          return (
            <div
              key={JSON.stringify(value)}
              style={{
                padding,
                fontSize,
                width: partialWidth,
                minWidth,
                backgroundColor: value.color
              }}>
              {value.amount}
            </div>
          )
        })
      }
    </div>
  )
}

export default PercentageBar