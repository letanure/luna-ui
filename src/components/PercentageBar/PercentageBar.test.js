import React from 'react'
import renderer from 'react-test-renderer'

import PercentageBar from './index'

describe('<PercentageBar />', () => {

  let component

  beforeEach(() => {
    const props = {
      fontSize: 12,
      padding: '3px',
      width: 300,
      values: [
        {
          amount: 745,
          color: '#b2e2bf'
        },
        {
          amount: 11036,
          color: '#c09be0'
        },
        {
          amount: 5160,
          color: '#c1e8f6'
        },
        {
          amount: 354,
          color: '#ffb61f'
        }
      ]
    }

    component = (
      <PercentageBar {...props} />
    )
  })

  test('toMatchSnapshot', () => {
    const tree = renderer
      .create(component)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

})
