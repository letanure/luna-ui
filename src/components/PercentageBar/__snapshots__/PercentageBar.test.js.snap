// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PercentageBar /> toMatchSnapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "width": 300,
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "#b2e2bf",
        "fontSize": 12,
        "minWidth": 30,
        "padding": "3px",
        "width": 12.92281006071119,
      }
    }
  >
    745
  </div>
  <div
    style={
      Object {
        "backgroundColor": "#c09be0",
        "fontSize": 12,
        "minWidth": 30,
        "padding": "3px",
        "width": 191.43104943625326,
      }
    }
  >
    11036
  </div>
  <div
    style={
      Object {
        "backgroundColor": "#c1e8f6",
        "fontSize": 12,
        "minWidth": 30,
        "padding": "3px",
        "width": 89.50563746747616,
      }
    }
  >
    5160
  </div>
  <div
    style={
      Object {
        "backgroundColor": "#ffb61f",
        "fontSize": 12,
        "minWidth": 30,
        "padding": "3px",
        "width": 6.14050303555941,
      }
    }
  >
    354
  </div>
</div>
`;
